src/legacy/core_plugins/kibana/migrations/types.ts

```diff
@@ -17,7 +17,8 @@
  * under the License.
  */
 
-import { SavedObjectReference } from '../../../../legacy/server/saved_objects/routes/types';
+// eslint-disable-next-line @kbn/eslint/no-restricted-paths
+import { SavedObjectReference } from '../../../../core/server';
```

Comment:
This is server code (SO migrations), but outside of the `server` folder, which explains the disabled rule